lib/intelligent_foods/api/v1.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module IntelligentFoods
+  class V1
+    attr_accessor :authentication
+    attr_reader :environment, :username, :password
+
+    PRODUCTION = "production"
+    STAGING = "staging"
+
+    def initialize(username: nil, password: nil, authentication: nil,
+                   environment: STAGING)
+      @authentication = authentication
+      @environment = environment
+      @username = username
+      @password = password
+    end
+
+    def base_url
+      "https://api.sunbasket.#{tld}/partner/v1"
+    end
+
+    def create_authentication!
+      authenticator = IntelligentFoods::Authenticator.build(self)
+      @authentication = authenticator.save!
+    end
+
+    def client
+      build_api_client!
+    end
+
+    def authenticated?
+      authentication.present?
+    end
+
+    def reset_authentication
+      @authentication = nil
+      self
+    end
+
+    def self.build
+      new(environment: IntelligentFoods.environment,
+          username: IntelligentFoods.client_id,
+          password: IntelligentFoods.client_secret)
+    end
```

Comment:
always prefer injection with defaults wherever possible

```suggestion
    def self.build(config: IntelligentFoods)
      new(environment: config.environment,
          username: config.client_id,
          password: config.client_secret)
    end
```